### PR TITLE
added weak scaling instructions

### PR DESCRIPTION
**Pull request by Yurlungur**
As per @gshipman 's request, this PR adds documentation for weak scaling the advection test. This also necessitated a tiny change to `parthinput.advection`, where I add `nlim=-1` to the `parthenon/time` block. This doesn't change anything for the regression tests, but it allows `nlim` to be modified on the command line.

I think it's worth documenting this stuff, though maybe it should go into the wiki instead of the docs. Interested in people's thoughts.

## PR Checklist

- [x] Code passes cpplint
- [x] New features are documented.
- [x] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md

### CODE REVIEW

**Comment by Yurlungur** [comment]
@AndrewGaspar @gshipman @jdolence @pgrete @forrestglines @nmsriram @JoshuaSBrown @jlippuner  I'd like to get this into develop if you all agree that's worth doing. Can a few people add your reviews? Or alternatively, object?

**Review by AndrewGaspar — APPROVED** [review]

File: docs/weak_scaling.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+git checkout develop && git pull
+git submodule update --init --recursive
+mkdir -p bin && cd bin
+cmake -DCMAKE_BUILD_TYPE=Release -DKokkos_ENABLE_OPENMP=True -DKokkos_ARCH_POWER9=True -DKokkos_ENABLE_CUDA=True -DKokkos_ARCH_VOLTA70=True -DCMAKE_CXX_COMPILER=${PWD}/../external/Kokkos/bin/nvcc_wrapper ..
+make -j
+```
+
+### To Run
+
+- Note that if you disabled HDF5 in the previous step, you mest open
```
Comment (brryan) [inline docs/weak_scaling.md]: mest -> must

**Review by brryan — APPROVED** [review]
Looks good to me -- caught one typo if you think it's worth changing

File: docs/weak_scaling.md (hunk 1 above)
Comment (Yurlungur) [reply docs/weak_scaling.md]: Oops good catch. I'll fix that!

**Pull request merged**